diff --color -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/CMakeLists.txt	2023-07-04 21:52:10.520708012 +0200
@@ -391,8 +391,8 @@
     set (HAVE_SPATIALITE TRUE)
   endif()
 
-  if (NOT PROJ_FOUND OR NOT GEOS_FOUND OR NOT GDAL_FOUND)
-    message (SEND_ERROR "Some dependencies were not found! Proj: ${PROJ_FOUND}, Geos: ${GEOS_FOUND}, GDAL: ${GDAL_FOUND}")
+  if (NOT Proj_FOUND OR NOT GEOS_FOUND OR NOT GDAL_FOUND)
+    message (SEND_ERROR "Some dependencies were not found! Proj: ${Proj_FOUND}, Geos: ${GEOS_FOUND}, GDAL: ${GDAL_FOUND}")
   endif()
 
   if (POSTGRES_FOUND)
diff --color -Naur a/external/qspatialite/CMakeLists.txt b/external/qspatialite/CMakeLists.txt
--- a/external/qspatialite/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/external/qspatialite/CMakeLists.txt	2023-06-06 02:14:26.789057740 +0200
@@ -5,20 +5,15 @@
 add_definitions(-DQT_PLUGIN)
 add_definitions(-DQT_SHARED)
 
-include_directories(SYSTEM
-	${SQLITE3_INCLUDE_DIR}
-	${Qt5Sql_PRIVATE_INCLUDE_DIRS}
-)
-
 set(QSQLSPATIALITE_SRC qsql_spatialite.cpp smain.cpp)
 QT5_WRAP_CPP(QSQLSPATIALITE_SRC qsql_spatialite.h smain.h)
 
 add_library(qsqlspatialite SHARED ${QSQLSPATIALITE_SRC})
 target_link_libraries(qsqlspatialite
-        ${Qt5Core_LIBRARIES}
-        ${Qt5Sql_LIBRARIES}
-	${SQLITE3_LIBRARY}
-	${SPATIALITE_LIBRARY}
+        ${QT_VERSION_BASE}::Core
+        ${QT_VERSION_BASE}::Sql
+	SQLite::SQLite3
+	spatialite::spatialite
 	qgis_core
 )
 
diff --color -Naur a/src/3d/CMakeLists.txt b/src/3d/CMakeLists.txt
--- a/src/3d/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/3d/CMakeLists.txt	2023-06-06 02:09:25.725648825 +0200
@@ -236,11 +236,9 @@
   ${CMAKE_SOURCE_DIR}/external/delaunator-cpp
 )
 
-if (BUILD_WITH_QT6)
-  target_link_libraries(qgis_3d Qt6::3DCore Qt6::3DRender Qt6::3DInput Qt6::3DLogic Qt6::3DExtras)
-else()
-  target_link_libraries(qgis_3d Qt5::3DCore Qt5::3DRender Qt5::3DInput Qt5::3DLogic Qt5::3DExtras)
-endif()
+
+target_link_libraries(qgis_3d ${QT_VERSION_BASE}::3DCore Qt6::3DRender ${QT_VERSION_BASE}::3DInput ${QT_VERSION_BASE}::3DLogic ${QT_VERSION_BASE}::3DExtras)
+
 
 GENERATE_EXPORT_HEADER(
    qgis_3d
diff --color -Naur a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
--- a/src/core/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/core/CMakeLists.txt	2023-06-06 01:14:33.090746702 +0200
@@ -2323,9 +2323,9 @@
   ${OPTIONAL_QTWEBKIT}
   ${QCA_LIBRARY}
   ${QTKEYCHAIN_LIBRARY}
-  ${PROJ_LIBRARY}
-  ${GEOS_LIBRARY}
-  ${GDAL_LIBRARY}
+  PROJ::proj
+  GEOS::geos
+  GDAL::GDAL
   ${SPATIALINDEX_LIBRARY}
   ${EXPAT_LIBRARY}
   ${SQLITE3_LIBRARY}
diff --color -Naur a/src/crssync/CMakeLists.txt b/src/crssync/CMakeLists.txt
--- a/src/crssync/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/crssync/CMakeLists.txt	2023-06-06 02:06:05.709990177 +0200
@@ -10,8 +10,8 @@
 
   target_link_libraries(crssync
     qgis_core
-    ${PROJ_LIBRARY}
-    ${GDAL_LIBRARY}
+    PROJ::proj
+    GDAL::GDAL
   )
 
   if(MSVC AND NOT USING_NMAKE)
diff --color -Naur a/src/native/CMakeLists.txt b/src/native/CMakeLists.txt
--- a/src/native/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/native/CMakeLists.txt	2023-06-06 01:53:15.018365720 +0200
@@ -134,12 +134,9 @@
   )
 endif()
 
-if (BUILD_WITH_QT6)
-  target_link_libraries(qgis_native ${Qt6Core_LIBRARIES} ${Qt6Gui_LIBRARIES} "${NATIVE_LINK_LIBS}")
-else()
-  target_link_libraries(qgis_native ${Qt5Core_LIBRARIES} ${Qt5Gui_LIBRARIES} "${NATIVE_LINK_LIBS}")
-endif()
-
+target_link_libraries(qgis_native ${QT_VERSION_BASE}::Core ${QT_VERSION_BASE}::Gui "${NATIVE_LINK_LIBS}")
+  
+  
 if (UNIX AND NOT APPLE AND NOT ANDROID)
   target_link_libraries(qgis_native ${QT_VERSION_BASE}::DBus)
 endif()
diff --color -Naur a/src/plugins/offline_editing/CMakeLists.txt b/src/plugins/offline_editing/CMakeLists.txt
--- a/src/plugins/offline_editing/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/plugins/offline_editing/CMakeLists.txt	2023-06-06 02:04:57.114093758 +0200
@@ -36,7 +36,7 @@
 target_link_libraries(plugin_offlineediting
   qgis_core
   qgis_gui
-  ${SPATIALITE_LIBRARY}
+  spatialite::spacialite
 )
 
 include_directories(SYSTEM
diff --color -Naur a/src/process/CMakeLists.txt b/src/process/CMakeLists.txt
--- a/src/process/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/process/CMakeLists.txt	2023-06-06 01:22:09.492593690 +0200
@@ -42,9 +42,9 @@
   qgis_core
   qgis_analysis
   ${QT_VERSION_BASE}::Core
-  ${PROJ_LIBRARY}
-  ${GEOS_LIBRARY}
-  ${GDAL_LIBRARY}
+  PROJ::proj
+  GEOS::geos
+  GDAL::GDAL
 )
 
 if (WITH_3D)
diff --color -Naur a/src/providers/arcgisrest/CMakeLists.txt b/src/providers/arcgisrest/CMakeLists.txt
--- a/src/providers/arcgisrest/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/providers/arcgisrest/CMakeLists.txt	2023-06-06 02:46:04.953908814 +0200
@@ -33,7 +33,7 @@
 
 target_link_libraries (provider_arcgisfeatureserver_a
   qgis_core
-  ${QCA_LIBRARY}
+  qca::qca
 )
 
 if (WITH_GUI)
@@ -74,7 +74,7 @@
   
   target_link_libraries(provider_arcgisfeatureserver
     qgis_core
-    ${QCA_LIBRARY}
+    qca::qca
   )
   
   # require c++17
@@ -110,7 +110,7 @@
 
 target_link_libraries (provider_arcgismapserver_a
   qgis_core
-  ${QCA_LIBRARY}
+  qca::qca
 )
 
 if (FORCE_STATIC_LIBS)
@@ -124,7 +124,7 @@
   
   target_link_libraries(provider_arcgismapserver
     qgis_core
-    ${QCA_LIBRARY}
+    qca::qca
   )
   
   install (TARGETS provider_arcgismapserver
diff --color -Naur a/src/providers/CMakeLists.txt b/src/providers/CMakeLists.txt
--- a/src/providers/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/providers/CMakeLists.txt	2023-06-06 03:00:27.092824043 +0200
@@ -6,7 +6,7 @@
 add_subdirectory(wms)
 add_subdirectory(wcs)
 add_subdirectory(delimitedtext)
-add_subdirectory(arcgisrest)
+###add_subdirectory(arcgisrest)
 
 if (WITH_SPATIALITE)
   add_subdirectory(spatialite)
@@ -21,7 +21,7 @@
 if (NOT FORCE_STATIC_LIBS)
   add_subdirectory(mssql)
   add_subdirectory(gpx)
-  add_subdirectory(mdal)
+###  add_subdirectory(mdal)
   if (WITH_ANALYSIS)
     add_subdirectory(virtualraster)
   endif()
diff --color -Naur a/src/providers/gpx/CMakeLists.txt b/src/providers/gpx/CMakeLists.txt
--- a/src/providers/gpx/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/providers/gpx/CMakeLists.txt	2023-06-06 01:39:08.872397282 +0200
@@ -34,7 +34,7 @@
 target_compile_features(provider_gpx PRIVATE cxx_std_17)
 
 target_link_libraries(provider_gpx
-  ${EXPAT_LIBRARY}
+  expat::expat
   qgis_core
 )
 
diff --color -Naur a/src/providers/grass/CMakeLists.txt b/src/providers/grass/CMakeLists.txt
--- a/src/providers/grass/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/providers/grass/CMakeLists.txt	2023-06-06 01:33:10.473829678 +0200
@@ -214,14 +214,14 @@
       target_link_libraries(qgis.d.rast${GRASS_BUILD_VERSION}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_gis}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_datetime}
-        ${GDAL_LIBRARY}
+        GDAL::GDAL
       )
     else()
       target_link_libraries(qgis.d.rast${GRASS_BUILD_VERSION}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_gis}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_datetime}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_raster}
-        ${GDAL_LIBRARY}
+	GDAL::GDAL
       )
     endif()
 
@@ -234,7 +234,7 @@
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_gis}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_datetime}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_gproj}
-        ${GDAL_LIBRARY}
+	GDAL::GDAL
       )
     else()
       target_link_libraries(qgis.g.info${GRASS_BUILD_VERSION}
@@ -242,7 +242,7 @@
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_datetime}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_gproj}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_raster}
-        ${GDAL_LIBRARY}
+	GDAL::GDAL
       )
     endif()
     if (UNIX)
@@ -261,7 +261,7 @@
         qgisgrass${GRASS_BUILD_VERSION}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_gis}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_datetime}
-        ${GDAL_LIBRARY}
+	GDAL::GDAL
         qgis_core
       )
     else()
@@ -270,7 +270,7 @@
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_gis}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_datetime}
         ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_raster}
-        ${GDAL_LIBRARY}
+	GDAL::GDAL
         qgis_core
       )
     endif()
@@ -289,7 +289,7 @@
       ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_vect}
       ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_dbmibase}
       ${GRASS_LIBRARY${GRASS_BUILD_VERSION}_dbmiclient}
-      ${GDAL_LIBRARY}
+      GDAL::GDAL
       qgis_core
     )
 
diff --color -Naur a/src/providers/hana/CMakeLists.txt b/src/providers/hana/CMakeLists.txt
--- a/src/providers/hana/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/providers/hana/CMakeLists.txt	2023-06-06 01:46:25.877044766 +0200
@@ -67,8 +67,6 @@
 )
 
 include_directories (SYSTEM
-  ${PROJ_INCLUDE_DIR}
-  ${GDAL_INCLUDE_DIR}
   ${ODBC_INCLUDE_DIRS}
   ${QCA_INCLUDE_DIR}
   ${QTKEYCHAIN_INCLUDE_DIR}
diff --color -Naur a/src/providers/mdal/CMakeLists.txt b/src/providers/mdal/CMakeLists.txt
--- a/src/providers/mdal/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/providers/mdal/CMakeLists.txt	2023-06-06 01:26:54.607371656 +0200
@@ -203,25 +203,19 @@
 endif()
 
 if (HDF5_FOUND)
-  target_include_directories(provider_mdal PRIVATE ${HDF5_INCLUDE_DIRS})
-  target_link_libraries(provider_mdal ${HDF5_C_LIBRARIES} )
-  target_compile_definitions(provider_mdal PRIVATE ${HDF5_DEFINITIONS})
+  target_link_libraries(provider_mdal PRIVATE hdf5::hdf5 )
 endif()
 
 if (GDAL_FOUND)
-  target_include_directories(provider_mdal PRIVATE ${GDAL_INCLUDE_DIR})
-  target_link_libraries(provider_mdal ${GDAL_LIBRARY} )
+  target_link_libraries(provider_mdal PRIVATE GDAL::GDAL )
 endif()
 
 if (NETCDF_FOUND)
-  target_include_directories(provider_mdal PRIVATE ${NETCDF_INCLUDE_DIR})
-  target_link_libraries(provider_mdal ${NETCDF_LIBRARY} )
+  target_link_libraries(provider_mdal PRIVATE netCDF::netcdf )
 endif()
 
 if (LIBXML2_FOUND)
-  target_include_directories(provider_mdal PRIVATE ${LIBXML2_INCLUDE_DIR})
-  target_link_libraries(provider_mdal ${LIBXML2_LIBRARIES} )
-  target_compile_definitions(provider_mdal PRIVATE ${LIBXML2_DEFINITIONS})
+  target_link_libraries(provider_mdal LibXml2::LibXml2 )
 endif()
 
 # clang-tidy
diff --color -Naur a/src/providers/oracle/CMakeLists.txt b/src/providers/oracle/CMakeLists.txt
--- a/src/providers/oracle/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/providers/oracle/CMakeLists.txt	2023-06-06 01:47:32.481395031 +0200
@@ -39,7 +39,6 @@
   ${CMAKE_BINARY_DIR}/src/ui
 )
 include_directories(SYSTEM
-  ${GEOS_INCLUDE_DIR}
   ${QT_QTSQL_INCLUDEDIR}
 )
 
diff --color -Naur a/src/providers/pdal/CMakeLists.txt b/src/providers/pdal/CMakeLists.txt
--- a/src/providers/pdal/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/providers/pdal/CMakeLists.txt	2023-06-06 02:32:07.618211682 +0200
@@ -134,7 +134,8 @@
 )
 
 target_link_libraries (untwine
-  ${PDAL_LIBRARIES}
+  PDAL::pdal_base
+  PDAL::pdal_util
   Threads::Threads
 )
 target_include_directories(untwine PRIVATE ${UNTWINE_INCLUDE_DIRS})
@@ -152,13 +153,15 @@
 
 target_link_libraries (provider_pdal
   qgis_core
-  ${PDAL_LIBRARIES}
+  PDAL::pdal_base
+  PDAL::pdal_util
   Threads::Threads
 )
 
 if (WITH_GUI)
   target_link_libraries (provider_pdal
-    ${PDAL_LIBRARIES}
+    PDAL::pdal_base
+    PDAL::pdal_util
     ${QT_VERSION_BASE}::Xml
     ${QT_VERSION_BASE}::Core
     ${QT_VERSION_BASE}::Svg
@@ -177,7 +180,8 @@
 target_compile_features(provider_pdal_a PRIVATE cxx_std_17)
 
 target_link_libraries (provider_pdal_a
-  ${PDAL_LIBRARIES}
+  PDAL::pdal_base
+  PDAL::pdal_util
   ${QT_VERSION_BASE}::Xml
   ${QT_VERSION_BASE}::Core
   ${QT_VERSION_BASE}::Svg
@@ -194,7 +198,8 @@
   target_compile_features(provider_pdal_gui_a PRIVATE cxx_std_17)
 
   target_link_libraries (provider_pdal_gui_a
-    ${PDAL_LIBRARIES}
+    PDAL::pdal_base
+    PDAL::pdal_util
     ${QT_VERSION_BASE}::Xml
     ${QT_VERSION_BASE}::Core
     ${QT_VERSION_BASE}::Svg
diff --color -Naur a/src/providers/postgres/CMakeLists.txt b/src/providers/postgres/CMakeLists.txt
--- a/src/providers/postgres/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/providers/postgres/CMakeLists.txt	2023-06-06 01:37:41.956742489 +0200
@@ -55,7 +55,7 @@
 
 target_link_libraries (provider_postgres_a
   qgis_core
-  ${POSTGRES_LIBRARY}
+  libpq::pq
 )
 
 if (WITH_GUI)
@@ -81,7 +81,7 @@
   target_link_libraries(provider_postgres_gui_a
     qgis_gui
     provider_postgres_a
-    ${POSTGRES_LIBRARY}
+    libpq::pq
   )
   add_dependencies(provider_postgres_gui_a ui)
 endif()
@@ -114,7 +114,7 @@
 target_link_libraries (provider_postgresraster_a
     qgis_core
     provider_postgres_a
-    ${POSTGRES_LIBRARY}
+
 )
 
 if (WITH_GUI)
@@ -154,11 +154,14 @@
 
   target_link_libraries(provider_postgres
     qgis_core
+    libpq::pq
     ${POSTGRES_LIBRARY}
   )
 
   target_link_libraries(provider_postgresraster
     qgis_core
+    libpq::pq
+
     ${POSTGRES_LIBRARY}
   )
 
diff --color -Naur a/src/providers/spatialite/CMakeLists.txt b/src/providers/spatialite/CMakeLists.txt
--- a/src/providers/spatialite/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/providers/spatialite/CMakeLists.txt	2023-06-06 01:43:53.387960374 +0200
@@ -38,7 +38,7 @@
 
 target_link_libraries(provider_spatialite_a
   qgis_core
-  ${SPATIALITE_LIBRARY}
+  libspatialite::libspatialite
 )
 
 # require c++17
@@ -95,7 +95,7 @@
   
   target_link_libraries(provider_spatialite
     qgis_core
-    ${SPATIALITE_LIBRARY}
+    libspatialite::libspatialite
   )
 
   if (WITH_GUI)
diff --color -Naur a/src/providers/virtual/CMakeLists.txt b/src/providers/virtual/CMakeLists.txt
--- a/src/providers/virtual/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/providers/virtual/CMakeLists.txt	2023-06-06 01:29:29.802727993 +0200
@@ -40,8 +40,8 @@
 
 target_link_libraries(provider_virtuallayer_a
   qgis_core
-  ${SQLITE3_LIBRARY}
-  ${SPATIALITE_LIBRARY}
+  SQLite::SQLite3
+  libspatialite::libspatialite
 )
 
 # require c++17
@@ -97,8 +97,8 @@
     qgis_core
     ${QT_VERSION_BASE}::Core
     ${QT_VERSION_BASE}::Widgets
-    ${SQLITE3_LIBRARY}
-    ${SPATIALITE_LIBRARY}
+    SQLite::SQLite3
+    libspatialite::libspatialite
   )
 
   if (WITH_GUI)
diff --color -Naur a/src/providers/wms/CMakeLists.txt b/src/providers/wms/CMakeLists.txt
--- a/src/providers/wms/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/providers/wms/CMakeLists.txt	2023-06-06 01:41:55.695737416 +0200
@@ -80,7 +80,7 @@
 
   target_link_libraries(provider_wms
     qgis_core
-    ${GDAL_LIBRARY}  # for OGR_G_CreateGeometryFromJson()
+    GDAL::GDAL  # for OGR_G_CreateGeometryFromJson()
   )
 
   if (WITH_GUI)
diff --color -Naur a/src/server/CMakeLists.txt b/src/server/CMakeLists.txt
--- a/src/server/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/src/server/CMakeLists.txt	2023-06-06 01:54:35.438446858 +0200
@@ -113,11 +113,11 @@
 
 target_link_libraries(qgis_server
   qgis_core
-  ${PROJ_LIBRARY}
+  PROJ::proj
   ${FCGI_LIBRARY}
   ${POSTGRES_LIBRARY}
-  ${GDAL_LIBRARY}
-  ${QCA_LIBRARY}
+  GDAL::GDAL
+  gca::gca
 )
 
 if (WITH_ANALYSIS)
diff --color -Naur a/tests/src/3d/sandbox/CMakeLists.txt b/tests/src/3d/sandbox/CMakeLists.txt
--- a/tests/src/3d/sandbox/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/tests/src/3d/sandbox/CMakeLists.txt	2023-06-06 01:18:06.025700844 +0200
@@ -37,9 +37,9 @@
   ${QT_VERSION_BASE}::Svg
   ${QT_VERSION_BASE}::Network
   ${QT_VERSION_BASE}::Test
-  ${PROJ_LIBRARY}
-  ${GEOS_LIBRARY}
-  ${GDAL_LIBRARY}
+  PROJ::proj
+  GEOS::geos
+  GDAL::GDAL
   ${QWT_LIBRARY}
   qgis_core
   qgis_3d
diff --color -Naur a/tests/src/providers/grass/CMakeLists.txt b/tests/src/providers/grass/CMakeLists.txt
--- a/tests/src/providers/grass/CMakeLists.txt	2023-04-28 14:11:34.000000000 +0200
+++ b/tests/src/providers/grass/CMakeLists.txt	2023-06-06 01:06:57.501022326 +0200
@@ -1,13 +1,8 @@
 include_directories(
   ${CMAKE_SOURCE_DIR}/src/providers/grass
 )
-include_directories(BEFORE SYSTEM
-  ${PROJ_INCLUDE_DIR}
-  ${GDAL_INCLUDE_DIR}
-)
 include_directories(SYSTEM
   ${POSTGRES_INCLUDE_DIR}
-  ${GEOS_INCLUDE_DIR}
 )
 
 
@@ -29,9 +24,9 @@
     ${QT_VERSION_BASE}::Core
     ${QT_VERSION_BASE}::Svg
     ${QT_VERSION_BASE}::Test
-    ${PROJ_LIBRARY}
-    ${GEOS_LIBRARY}
-    ${GDAL_LIBRARY}
+    PROJ::proj
+    GEOS::geos
+    GDAL::GDAL
     qgis_core
     qgis_test
     qgisgrass${grass_build_version}
