diff --color -Naur a/cmake/Modules/Findlibsecret.cmake b/cmake/Modules/Findlibsecret.cmake
--- a/cmake/Modules/Findlibsecret.cmake	1970-01-01 01:00:00.000000000 +0100
+++ b/cmake/Modules/Findlibsecret.cmake	2023-07-06 11:35:09.791051085 +0200
@@ -0,0 +1,13 @@
+
+include(FindPkgConfig)
+
+set(libsecret_pkg_check_modules_OPTIONS)
+if(libsecret_FIND_REQUIRED)
+    set(libsecret_pkg_check_modules_OPTIONS REQUIRED)
+endif()
+
+pkg_check_modules(libsecret ${libsecret_pkg_check_modules_OPTIONS} IMPORTED_TARGET libsecret-1)
+
+if(libsecret_FOUND)
+    add_library(ALIAS PkgConfig::libsecret libsecret::libsecret)
+endif()
diff --color -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2023-06-01 13:38:35.000000000 +0200
+++ b/CMakeLists.txt	2023-07-06 11:36:12.314751752 +0200
@@ -8,7 +8,6 @@
 # Enable C++11
 SET(CMAKE_CXX_STANDARD 11)
 
-include(FindPkgConfig)
 
 ###
 
@@ -51,24 +50,23 @@
     find_package(Qt5 COMPONENTS Core REQUIRED)
 endif()
 
-if (Qt5Core_FOUND AND NOT BUILD_WITH_QT6)
+if (Qt5_FOUND AND NOT BUILD_WITH_QT6)
   set(QTKEYCHAIN_VERSION_INFIX 5)
 
   if(ANDROID)
     if(Qt5Core_VERSION VERSION_LESS 5.7)
         find_package(Qt5 COMPONENTS Core REQUIRED Private)
-        include_directories(${Qt5Core_PRIVATE_INCLUDE_DIRS})
+        list(APPEND qtkeychain_LIBRARIES Qt5::CorePrivate)
     endif()
 
     find_package(Qt5 COMPONENTS AndroidExtras REQUIRED)
-    include_directories(${Qt5AndroidExtras_INCLUDE_DIRS})
+    list(APPEND qtkeychain_LIBRARIES Qt5::AndroidExtras)
     set(QTANDROIDEXTRAS_LIBRARIES ${Qt5AndroidExtras_LIBRARIES})
   endif()
 
   if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT HAIKU)
     find_package(Qt5 COMPONENTS DBus REQUIRED)
-    include_directories(${Qt5DBus_INCLUDE_DIRS})
-    set(QTDBUS_LIBRARIES ${Qt5DBus_LIBRARIES})
+    list(APPEND qtkeychain_LIBRARIES Qt5::DBus)
     macro(qt_add_dbus_interface)
       qt5_add_dbus_interface(${ARGN})
     endmacro()
@@ -88,8 +86,7 @@
     qt5_wrap_cpp(${ARGN})
   endmacro()
 
-  set(QTCORE_LIBRARIES ${Qt5Core_LIBRARIES})
-  include_directories(${Qt5Core_INCLUDE_DIRS})
+  list(APPEND qtkeychain_LIBRARIES Qt5::Core)
 else()
   find_package(Qt6 COMPONENTS Core REQUIRED)
   set(QTKEYCHAIN_VERSION_INFIX 6)
@@ -97,8 +94,7 @@
 
   if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT HAIKU)
     find_package(Qt6 COMPONENTS DBus REQUIRED)
-    include_directories(${Qt6DBus_INCLUDE_DIRS})
-    set(QTDBUS_LIBRARIES ${Qt6DBus_LIBRARIES})
+    list(APPEND qtkeychain_LIBRARIES Qt5::DBus)
     macro(qt_add_dbus_interface)
       qt6_add_dbus_interface(${ARGN})
     endmacro()
@@ -118,7 +114,7 @@
     qt6_wrap_cpp(${ARGN})
   endmacro()
 
-  set(QTCORE_LIBRARIES ${Qt6Core_LIBRARIES})
+  list(APPEND qtkeychain_LIBRARIES Qt5::Core)
 endif()
 
 
@@ -172,22 +168,18 @@
     option(LIBSECRET_SUPPORT "Build with libsecret support" ON)
 
     if(LIBSECRET_SUPPORT)
-        pkg_check_modules(LIBSECRET REQUIRED libsecret-1)
+        find_package(libsecret REQUIRED)
         add_definitions(-DHAVE_LIBSECRET=1)
-        INCLUDE_DIRECTORIES(${LIBSECRET_INCLUDE_DIRS})
-        LINK_DIRECTORIES(${LIBSECRET_LIBRARY_DIRS})
-        list(APPEND qtkeychain_LIBRARIES_PRIVATE ${LIBSECRET_LIBRARIES})
+        list(APPEND qtkeychain_LIBRARIES_PRIVATE libsecret::libsecret)
     endif()
 
     add_definitions(-DKEYCHAIN_DBUS=1)
     list(APPEND qtkeychain_SOURCES keychain_unix.cpp gnomekeyring.cpp libsecret.cpp plaintextstore.cpp)
     qt_add_dbus_interface(qtkeychain_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/org.kde.KWallet.xml kwallet_interface KWalletInterface)
-    list(APPEND qtkeychain_LIBRARIES ${QTDBUS_LIBRARIES} )
 endif()
 
 if(ANDROID)
     list(APPEND qtkeychain_SOURCES keychain_android.cpp androidkeystore.cpp plaintextstore.cpp)
-    list(APPEND qtkeychain_LIBRARIES_PRIVATE ${QTANDROIDEXTRAS_LIBRARIES} )
 endif()
 
 QT_WRAP_CPP(qtkeychain_MOC_OUTFILES keychain.h keychain_p.h gnomekeyring_p.h)
@@ -237,7 +229,7 @@
 
         install(FILES ${qtkeychain_QM_FILES} DESTINATION ${QTKEYCHAIN_TRANSLATIONS_DIR})
     endif()
-endif( BUILD_TRANSLATIONS )
+endif()
 
 target_link_libraries(${QTKEYCHAIN_TARGET_NAME} PUBLIC ${qtkeychain_LIBRARIES} PRIVATE ${qtkeychain_LIBRARIES_PRIVATE})
 if(NOT INTERFACE_INCLUDE_SUFFIX)
diff --color -Naur a/keychain.h b/keychain.h
--- a/keychain.h	2023-06-01 13:38:35.000000000 +0200
+++ b/keychain.h	2023-07-06 11:19:13.443107036 +0200
@@ -277,6 +277,6 @@
  */
 QKEYCHAIN_EXPORT bool isAvailable();
 
-} // namespace QtKeychain
+} // namespace QKeychain
 
 #endif
